pkg/build/api/validation/validation.go

```diff
@@ -31,7 +31,7 @@ func ValidateBuildUpdate(build *buildapi.Build, older *buildapi.Build) fielderro
 	allErrs = append(allErrs, ValidateBuild(build)...)
 
 	if !kapi.Semantic.DeepEqual(build.Spec, older.Spec) {
-		allErrs = append(allErrs, fielderrors.NewFieldInvalid("spec", build.Spec, "spec is immutable"))
+		allErrs = append(allErrs, fielderrors.NewFieldInvalid("spec", "build "+build.ObjectMeta.Name, "spec is immutable"))
```

Comment:
@liggitt bothers me that that message doesn't tell you which pod the spec is from...  is it safe to assume there will always be enough surrounding context to tell the user which pod they were trying to modify?  (and that it was a pod, for that matter)?

if so i'm fine w/ that message.